allennlp/data/dataset_readers/wikitables/world.py

```diff
@@ -0,0 +1,111 @@
+        # TODO: Should we do updates while reading tables instead?
+        self.name_mapping = {"reverse": "R", "max": "M0", "min": "M1",
+                             "argmax": "A0", "argmin": "A1",
+                             "and": "A", "or": "O", "next": "N",
+                             "fb:cell.cell.date": "D", "fb:cell.cell.number": "B"}
+        # For every new Sempre column name seen, we update this counter to map it to a new NLTK name.
+        self._column_counter = 0
+        # Initial type signature. Will update when we see more predicates.
+        self.type_signature = {"R": WTD.REVERSE_TYPE, "M0": WTD.MAX_MIN_TYPE, "M1": WTD.MAX_MIN_TYPE,
+                               "A0": WTD.MAX_MIN_TYPE, "A1": WTD.MAX_MIN_TYPE}
+        self._logic_parser = LogicParser(type_check=True)
+
+    def process_sempre_forms(self, sempre_forms: List[str]) -> List[Expression]:
+        """
+        Processes logical forms in Sempre format (that are either gold annotations or output from DPD)
+        and converts them into ``Expression`` objects in NLTK. The conversion involves mapping names to
+        follow NLTK's variable naming conventions.
+        """
+        expressions = []
+        for sempre_form in sempre_forms:
+            parsed_lisp = OneOrMore(nestedExpr()).parseString(sempre_form).asList()
+            translated_string = self._process_nested_expression(parsed_lisp)
+            expression = self._logic_parser.parse(translated_string, signature=self.type_signature)
+            expressions.append(expression)
+        return expressions
+
+    def _process_nested_expression(self, nested_expression):
+        """
+        ``nested_expression`` is the result of parsing a Lambda-DCS expression in Lisp format. We process it
+        recursively and return a string in the format that NLTK's ``LogicParser`` would understand.
+        """
+        if isinstance(nested_expression, list) and len(nested_expression) == 1:
+            return self._process_nested_expression(nested_expression[0])
+        elements_are_leaves = [isinstance(element, str) for element in nested_expression]
+        if all(elements_are_leaves):
+            mapped_names = [self._map_name(name) for name in nested_expression]
+        else:
+            mapped_names = []
+            for element, is_leaf in zip(nested_expression, elements_are_leaves):
+                if is_leaf:
+                    mapped_names.append(self._map_name(element))
+                else:
+                    mapped_names.append(self._process_nested_expression(element))
+        return "(%s %s)" % (mapped_names[0], " ".join("(%s)" % name for name in mapped_names[1:]))
+
+    def _map_name(self, sempre_name: str):
+        """
+        Takes the name of a predicate or a constant as used by Sempre, maps it to a unique string such that
+        NLTK processes it appropriately. This is needed because NLTK has a naming convention for variables:
```

Comment:
Does NLTK have processing logic that switches on this naming convention?

Reply:
https://github.com/nltk/nltk/blob/develop/nltk/sem/logic.py#L1817-L1848